app/services/parent_profile_associator.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+# A service class to associate profiles to their parent profiles
+class ParentProfileAssociator
+  class << self
+    def run!
+      Profile.transaction do
+        Profile.where.not(account: nil).find_each do |profile|
+          profile.update!(parent_profile: find_parent(profile))
+        end
+      end
+    end
+
+    private
+
+    def find_parent(profile)
+      parents = Profile.where(ref_id: profile.ref_id,
```

Comment:
Just a nitpick, why not `Profile.find` so you just get 1 result?

Reply:
I want to raise an error if the response is not 1 record. `find` only accepts an ID, not ref_id, account_id, and benchmark_id. `find` also only raises an error if no records are found; `find_by` is the same. Neither raises an error if multiple records are found.